api/support/api.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package support contains the support service API client.
+package support
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/juju/errors"
+	"github.com/juju/romulus/wireformat/support"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+	"gopkg.in/macaroon.v1"
+)
+
+var DefaultURL = "https://api.jujucharms.com/omnibus/v2"
+
+// SupportAuthClient defines the interface available to clients of the support api.
+type SupportAuthClient interface {
+	// Authorize returns the support macaroon for the specified model
+	Authorize(modelUUID, supportLevel string) (*macaroon.Macaroon, error)
+}
+
+var _ SupportAuthClient = (*client)(nil)
+
+type httpClient interface {
+	// Do sends the given HTTP request and returns its response.
+	Do(*http.Request) (*http.Response, error)
```

Comment:
we don't need Do atm.. we only do a POST, which uses DoWithBody.. min interfaces, etc..